File: app/src/main/java/com/example/user/uberuber/Remotes/RetrofitClient.java
package com.example.user.uberuber.Remotes;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.converter.scalars.ScalarsConverterFactory;

/**
 * Created by User on 7/4/2018.
 */

public class RetrofitClient {
    private static Retrofit retrofit = null;
    public static Retrofit getClient(String baseURL)
    {
        if (retrofit == null)
        {
            retrofit = new Retrofit.Builder()
                    .baseUrl(baseURL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

        }

        return retrofit;
    }

}

File: app/src/main/java/com/example/user/uberuber/Remo/RetrofitClient.kt
package com.example.user.uberuber.Remo

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by User on 17/4/2018.
 */
object RetrofitClient {
    private var retrofit: Retrofit?=null

    fun getClient(baseUrl: String): Retrofit {
        if (retrofit == null)
        {
            retrofit = Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

        }
        return retrofit!!
    }
}